### PR TITLE
ci: introduce license whitelist and add a mandatory license check

### PR DESCRIPTION
**Pull request by PhilipGough**
To test this, run `license_finder --decisions-file=.dependency_decisions.yml`

@unleashed looking at our ci to see how to integrate this as I'm not familiar with Travis

### CODE REVIEW

**Comment by PhilipGough** [comment]
I've created an issue in license_finder (see text in yams)

File: ci/script.sh [hunk 1]
```diff
@@ -14,6 +14,11 @@ main() {
     cross test --target $TARGET
     cross test --target $TARGET --release
 
+    ruby --version
+    ls -la
+    which license_finder
```
Comment (unleashed) [inline ci/script.sh]: guess these 3 commands were a test and can be removed?

Comment (PhilipGough) [reply ci/script.sh]: hmm, that seems to be outdated but let me squash the code changes and remove that entry correctly.

**Pull request merged**